test/cmd/diagnostics.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+OS_ROOT=$(dirname "${BASH_SOURCE}")/../..
+source "${OS_ROOT}/hack/util.sh"
+source "${OS_ROOT}/hack/cmd_util.sh"
+os::log::install_errexit
+
+# This test validates the diagnostics command
+
+# available diagnostics (2015-12-21):
+# AnalyzeLogs ClusterRegistry ClusterRoleBindings ClusterRoles ClusterRouter ConfigContexts DiagnosticPod MasterConfigCheck MasterNode NodeConfigCheck NodeDefinitions UnitStatus
+# Without things feeding into systemd, AnalyzeLogs and UnitStatus are irrelevant.
+# ConfigContexts runs into some invalid contexts created by other tests, so is omitted.
```

Comment:
Specifically what is being generated by other tests that is ruining a clean diagnostics run? Projects we create in other tests should be gone without a trace by the time this test runs around, so their existence is a bug. It's possible what you want to "clean" is outside the scope of what `test/cmd` buckets should be able to expect, but it may be that our framework isn't living up to the design.

Since your tests seem to require a node for some diagnostics, it makes more sense to stick this into an end-to-end bucket anyway where you can manually set up a server and be sure nobody else will ever touch it.